### PR TITLE
Asset replace

### PR DESCRIPTION
**Pull request by vkalpias**
This PR adds functionality to replace references to an asset with another asset. It also adds API for scene settings which is needed in part for the replacing mechanism which has to update the scene's skybox when a cubemap is replaced.

### CODE REVIEW

File: docs/classes/SceneSettingsProperties.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+| `physics.gravity` | `Array<number>` | An array of 3 numbers that represents the gravity force. |  | 
+| `render` | `object` | Render settings for the scene. |  | 
+| `render.exposure` | `number` | The exposure value tweaks the overall brightness of the scene. |  | 
+| `render.fog` | `string` | The type of fog used in the scene. Can be one of `pc.FOG_NONE`, `pc.FOG_LINEAR`, `pc.FOG_EXP`, `pc.FOG_EXP2`. |  | 
+| `render.fog_color` | `Array<number>` | An array of 3 numbers representing the color of the fog. |  | 
+| `render.fog_density` | `number` | The density of the fog. This property is only valid if the fog property is set to `pc.FOG_EXP` or `pc.FOG_EXP2`. |  | 
+| `render.fog_end` | `number` | The distance from the viewpoint where linear fog reaches its maximum. This property is only valid if the fog property is set to `pc.FOG_LINEAR`. |  | 
+| `render.fog_start` | `number` | The distance from the viewpoint where linear fog begins. This property is only valid if the fog property is set to `pc.FOG_LINEAR`. |  | 
+| `render.gamma_correction` | `number` | The gamma correction to apply when rendering the scene. Can be one of `pc.GAMMA_NONE`, `pc.GAMMA_SRGB`. |  |
```
Comment (willeastcott) [inline docs/classes/SceneSettingsProperties.md:18]: The properties with underscores really stand out. Can we alias these somehow to make them compatible with our camelCase standard?

File: src/asset.js [hunk 2]
```diff
@@ -275,6 +276,18 @@ class Asset extends Events {
         return entities[0];
     }
 
+    /**
+     * Replaces any references to this asset with
+     * references to the new asset specified.
+     *
+     * @param {Asset} asset - The new asset.
+     * @param {object} options - Options.
```
Comment (willeastcott) [inline src/asset.js:284]: What are the options?

**Review by willeastcott — APPROVED** [review]

**Comment by willeastcott** [comment]
I must say, it's pretty hard/inefficient to review these PRs. So many files with changes that are irrelevant to the review.

File: docs/classes/SceneSettingsProperties.md (hunk 1 above)
Comment (vkalpias) [reply docs/classes/SceneSettingsProperties.md:18]: Not easily

**Comment by vkalpias** [comment]
> I must say, it's pretty hard/inefficient to review these PRs. So many files with changes that are irrelevant to the review.

Do you not use the filter button?

**Pull request merged**